teku/src/main/java/tech/pegasys/teku/cli/subcommand/ValidatorApiClientUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.cli.subcommand;
+
+import static tech.pegasys.teku.infrastructure.logging.SubCommandLogger.SUB_COMMAND_LOG;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpClientAuthLoggingIntercepter;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpValidatorRestApiClient;
+
+class ValidatorApiClientUtil {
```

Comment:
I suspect we can avoid the mockito-inline dependency by making these methods non-static and injecting an implementation of this class into constructors of objects that need it.

I'd suggest it probably gets renamed to something like `RemoteSpecLoader` as well?

Reply:
hmm, looking at the specifics of the test I can see where the challenge comes from - `ValidatorClientCommand` is being created automatically by `PicoCLI`.  Two options I can see:

1. Inject this class into `BeaconNodeCommand` and then `ValidatorClientCommand` can access it via the `parentCommand` reference it has.  Bit ugly but would work.
2. We could change the test to use the `MockServer` extension so it can use the real http library (`ExternalSignerIntegrationTest` is an example of using this). Makes it more of an integration test than a unit test but does then cover more of the code which isn't bad.

Reply:
>hmm, looking at the specifics of the test I can see where the challenge comes from - ValidatorClientCommand is being created automatically by PicoCLI.

exactly. I did consider the above options as well and decided against them at the time for the reasons you mentioned, but 2 is probably a much better alternative to mocking a static class and introducing mockito-inline .. will change it